### PR TITLE
Remove the Arcsight module and the modules framework

### PR DESCRIPTION
**Pull request by kaisecheng**
## Release notes

Remove Arcsight module

## What does this PR do?

Remove all module related code
- remove arcsight module
- remove module framework
- remove module tests
- remove module configs

The removal of doc is addressed in https://github.com/elastic/logstash/pull/16796

## Why is it important/What is the impact to the user?

All modules are removed in v9. 

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files (and/or docker env variables)
- [x] I have added tests that prove my fix is effective or that my feature works

## Author's Checklist

- [x] passed exhaustive [test](https://buildkite.com/elastic/logstash-exhaustive-tests-pipeline/builds/1033)
- [x] tested legacy monitoring
- [x] tested central pipeline management
- [x] `bin/logstash --modules arcsight --setup` gives `ERROR: Unrecognised option '--modules'`
- [x] Config module in logstash.yml gives `Your settings are invalid. Reason: Setting "modules" doesn't exist`
  ```
  modules:
  - name: arcsight
  ```  
  

## Related issues

- closes https://github.com/elastic/logstash/issues/16740

### CODE REVIEW

**Comment by kaisecheng** [comment]
green exhaustive [tests](https://buildkite.com/elastic/logstash-exhaustive-tests-pipeline/builds/1033)

File: docker/data/logstash/env2yaml/env2yaml.go
Comment (donoghuc) [inline docker/data/logstash/env2yaml/env2yaml.go:20]: Why was this moved? Doing the imports in alphabetical order seems like a pattern we want to keep.

File: docker/data/logstash/env2yaml/env2yaml.go
Comment (donoghuc) [inline docker/data/logstash/env2yaml/env2yaml.go:129]: Are these "cloud" settings related to the module being removed?

Comment (donoghuc) [reply docker/data/logstash/env2yaml/env2yaml.go:129]: I see these are removed in i18n etc too.

**Review by donoghuc — APPROVED** [review]
Looks like a clean removal. `+28 −7,374` is pretty nice!

File: config/logstash.yml
Comment (robbavey) [inline config/logstash.yml:202]: The `Cloud Settings` below still refer to module specific settings `var.elasticsearch.hosts` and `var.kibana.host`, etc. 
If these settings are not relevant at all, we should remove them. If they are relevant to central management and internal collection, we should move and rework them

File: docker/data/logstash/env2yaml/env2yaml.go
Comment (robbavey) [reply docker/data/logstash/env2yaml/env2yaml.go:129]: Is `cloud.id` here _only_ relevant to modules, or is this where it is also set for central management and internal collection?

**Comment by robbavey** [comment]
I think there are some additional removals - 

I don't believe we use this [ElasticsearchClient](https://github.com/elastic/logstash/blob/e1f4e772dc220f9582d8fd736d8f6d5d4e66aba3/logstash-core/lib/logstash/elasticsearch_client.rb) outside of Modules code, despite it living outside of the main modules code area... 
Once that has gone, I believe [this](https://github.com/elastic/logstash/blob/main/logstash-core/lib/logstash/util/manticore_ssl_config_helper.rb) can go too, along with
https://github.com/elastic/logstash/blob/main/logstash-core/lib/logstash/util/modules_setting_array.rb

File: docker/data/logstash/env2yaml/env2yaml.go
Comment (kaisecheng) [reply docker/data/logstash/env2yaml/env2yaml.go:129]: cloud.id is only used for modules
CPM uses `xpack.management.elasticsearch.cloud_id`. Legacy monitoring uses `xpack.monitoring.elasticsearch.cloud_id`

File: config/logstash.yml
Comment (kaisecheng) [reply config/logstash.yml:202]: deleted

**Comment by kaisecheng** [comment]
@robbavey this is ready for a second look
cloud.id is for modules only. I have removed the mentioned dangling code.

File: logstash-core/spec/logstash/webserver_spec.rb
Comment (robbavey) [inline logstash-core/spec/logstash/webserver_spec.rb:24]: Out of curiosity, do you know what changed that forced this change? Is it part of this change? Or something outside of it?

**Review by robbavey — COMMENTED** [review]
This is so awesome! I do have a couple of questions before approval:

* Do we know *why* those two `require` statements suddenly needed to be added? Was it related to this PR, or a general issue?
* It didn't look like it from reading through the code, but did we confirm whether or not there are dependencies that we import that we no longer need to after this PR?

File: logstash-core/spec/logstash/webserver_spec.rb
Comment (kaisecheng) [reply logstash-core/spec/logstash/webserver_spec.rb:24]: This is caused by the removal of [elasticsearch_client.rb](https://github.com/elastic/logstash/pull/16794/commits/b47188abdf1c626c1b132c13d536635180f4d805#diff-df536490516d7a5ffd225497283c381778be1718b2d7b9b285721c65426fc027L18-L19) and the import in [runner_spec.rb](https://github.com/elastic/logstash/pull/16794/commits/b47188abdf1c626c1b132c13d536635180f4d805#diff-a4e3e6b841aa2183c5550359b0732036b0f8db6f1690434058be301416a49908L25). The elasticsearch related classes somehow include Faraday.

The import in other spec files affect the global namespace for the entire test suite. You can add `require "faraday"` in runner_spec and get a green run of `./gradlew :logstash-core:rubyTests`, but it will fail when running a single rspec file as webserver_spec itself doesn't link faraday.

**Review by robbavey — APPROVED** [review]
LGTM!

**Pull request merged**
